### PR TITLE
Performance improvement of Julia implementation 

### PR DESCRIPTION
**Pull request by doraemonho**
Hi,

Just make some changes to the Julia implementation. I choose the create a new jl file as I heavily change the code structure with 2 different approaches, while the performances are similar.

The new implementation should now more align with the julia's standard coding style and more importantly, improving the performance of Julia by an order of magnitude. Now Julia should be concise (<30 lines of main code) and comparable to the performance of C++.
The following plot is performed under NERSC's Perlmutter CPU node with `julia. 1.9.0-rc2`
image

Changes are:
1. Rewrite of `benchmark` function
2. Rewrite of `work` function
2.1 Polyester's approach to compute the heat equation (`New_heat2.jl`)
2.2 FastBroadcast's approach to compute the heat equation (`New_heat.jl`)

### CODE REVIEW

**Comment by doraemonho** [comment]
Hi @diehlpk , may I know any update on this?

**Comment by diehlpk** [comment]
Sorry, I was at conferences and just saw the pull request. I will have a look next week. 

I will definitely mention the new Julia code in my talk.

File: julia/heat/New_heat.jl
Comment (diehlpk) [inline julia/heat/New_heat.jl:1]: Please add your name and email address to the authors file to get credit for your good work.

https://github.com/diehlpk/async_heat_equation/blob/main/AUTHORS

File: julia/heat/New_heat.jl [hunk 1]
```diff
@@ -0,0 +1,55 @@
+using FastBroadcast
```
Comment (diehlpk) [inline julia/heat/New_heat.jl:7]: Can you please add the license on top of the file?

https://github.com/diehlpk/async_heat_equation/blob/df43c66d0eec780748b92f1232ec4746d7a53ad7/julia/heat/heat_ghosts.jl#L1

File: julia/heat/New_heat.jl [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+using BenchmarkTools
```
Comment (diehlpk) [inline julia/heat/New_heat.jl:8]: Can you please add the license to this file?

https://github.com/diehlpk/async_heat_equation/blob/df43c66d0eec780748b92f1232ec4746d7a53ad7/julia/heat/heat_ghosts.jl#L1

**Review by diehlpk — COMMENTED** [review]
@doraemonho it would be nice if you could fix the two minor issues below.

After that I am happy to approve your pull request.

**Comment by doraemonho** [comment]
@diehlpk Thanks for checking! I just make changes according to your comment.
Let me know if I miss anything

**Review by diehlpk — APPROVED** [review]
LGTM!

**Pull request merged**